pyteal/ast/itxn.py

```diff
@@ -200,6 +206,111 @@ def SetFields(cls, fields: Dict[TxnField, Union[Expr, List[Expr]]]) -> Expr:
         fieldsToSet = [cls.SetField(field, value) for field, value in fields.items()]
         return Seq(fieldsToSet)
 
+    @classmethod
+    def MethodCall(
+        cls,
+        *,
+        app_id: Expr,
+        method_signature: str,
+        args: List[Union[abi.BaseType, Expr, Dict[TxnField, Union[Expr, List[Expr]]]]],
+        fields: Dict[TxnField, Union[Expr, List[Expr]]],
+    ) -> Expr:
+
+        # Default, always need these
+        fields_to_set = [
+            cls.SetField(TxnField.type_enum, TxnType.ApplicationCall),
+            cls.SetField(TxnField.application_id, app_id),
+        ]
+
+        # We only care about the args
+        arg_type_specs, _ = type_specs_from_signature(method_signature)
+
+        # Start app args with the method selector
+        app_args: List[Expr] = [MethodSignature(method_signature)]
+
+        # Transactions are not included in the App Call
+        txns_to_pass: List[Expr] = []
+
+        # Reference Types are treated specially
+        accts: List[Expr] = []
+        apps: List[Expr] = []
+        assets: List[Expr] = []
+
+        # Reference types are encoded as a uint8 index into
+        # their respective arrays
+        def get_uint8(i):
```

Comment:
Feel free to ignore. Seems like this would be useful helper function. Maybe a class method of `class Int`? This may be shot done by others as it seems to break our typical patterns, but here's what I could see:

```python
class Int(LeafExpr):
    ...

    @classmethod
    def as_uint8(cls, i: int) -> Expr:
        return Suffix(Itob(cls(i)), Int(7))
```

Reply:
or maybe just an instance method:

```python
class Int(LeafExpr):
    ...

    def as_uint8(self) -> Expr:
        return Suffix(Itob(self), Int(7))
```

Reply:
I'd almost prefer to put this kind of thing as a class method on the ABI type itself.

something like:
```py
def Uint(BaseType):
    ...
    def to_encoded(cls, i: Int)->Expr:
        return Suffix(Itob(i), Int(8-cls.byte_length_static()))
```

Or tweaking/reimplementing the uint_* helper methods in uint.py to not need some scratch var

Reply:
I don't have a strong opinion on this, so leave at your discretion. Others might have stronger opinions though.